darshan-util/pydarshan/tests/test_plot_access_histogram.py

```diff
@@ -0,0 +1,107 @@
+
+darshan.enable_experimental()
+
+
+class TestPlotAccessHistogram:
+    @pytest.mark.parametrize(
+        "log_path, mod",
+        [
+            ("examples/example-logs/dxt.darshan", "POSIX"),
+            ("examples/example-logs/ior_hdf5_example.darshan", "POSIX"),
+            ("examples/example-logs/ior_hdf5_example.darshan", "MPI-IO"),
+            ("examples/example-logs/sample-badost.darshan", "POSIX"),
+            ("examples/example-logs/sample.darshan", "POSIX"),
+            ("examples/example-logs/sample.darshan", "MPI-IO"),
+        ],
+    )
+    def test_xticks_and_labels(self, log_path, mod):
+        # check the x-axis tick mark locations and
+        # labels for `plot_access_histogram()`
+
+        report = darshan.DarshanReport(log_path)
+        fig = plot_access_histogram(report=report, mod=mod)
+
+        # retrieve the x-axis tick mark locations and labels
+        # from the output figure object
+        ax = fig.axes[0]
+        actual_xticks = ax.get_xticks()
+        actual_xticklabels = [tl.get_text() for tl in ax.get_xticklabels()]
+
+        # the expected x-axis tick marks and locations
+        # should be the same for all cases
+        expected_xticks = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+        expected_xticklabels = [
+            "0-100", "101-1K", "1K-10K", "10K-100K", "100K-1M",
+            "1M-4M", "4M-10M", "10M-100M", "100M-1G", "1G+",
+        ]
+        assert_array_equal(actual_xticks, expected_xticks)
+        assert_array_equal(actual_xticklabels, expected_xticklabels)
+
+    @pytest.mark.parametrize(
+        "log_path, mod, expected_rd_vals, expected_wr_vals",
+        [
+            (
+                "examples/example-logs/dxt.darshan",
+                "POSIX",
+                [2289, 1193, 2470, 173, 1, 0, 0, 0, 0, 0],
+                [160, 12, 1325, 0, 0, 0, 0, 0, 0, 0],
+            ),
+            (
+                "examples/example-logs/ior_hdf5_example.darshan",
+                "POSIX",
+                [3, 17, 0, 0, 16, 0, 0, 0, 0, 0],
+                [3, 4, 0, 0, 16, 0, 0, 0, 0, 0],
+            ),
+            (
+                "examples/example-logs/ior_hdf5_example.darshan",
+                "MPI-IO",
+                [3, 17, 0, 0, 16, 0, 0, 0, 0, 0],
+                [3, 4, 0, 0, 16, 0, 0, 0, 0, 0],
+            ),
+            (
+                "examples/example-logs/sample-badost.darshan",
+                "POSIX",
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 131072, 0, 0, 0, 0],
+            ),
+            (
+                "examples/example-logs/sample.darshan",
+                "POSIX",
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [4, 14, 0, 0, 0, 0, 0, 0, 16384, 0],
+            ),
+            (
+                "examples/example-logs/sample.darshan",
+                "MPI-IO",
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [4, 14, 0, 0, 0, 0, 0, 0, 16384, 0],
```

Comment:
hmm, duplicating the same file with different module and identical expected result is perhaps a bit awkward--are the two modules guaranteed to always have the same result for this function?

If that isn't necessarily a guarantee that might be a dataset you don't have access to in the main repo--I know you mentioned having all that you needed in the main repo, so maybe the guarantee is true?

Reply:
> hmm, duplicating the same file with different module and identical expected result is perhaps a bit awkward--are the two modules guaranteed to always have the same result for this function?

No, I just happened to use ones that have the same result for both modules. I'll make sure to change that, it looks like `tests/input/sample-dxt-simple.darshan` has a slight difference between `POSIX` and `MPIIO`. Additionally `skew-app.darshan` and `imbalanced-io.darshan` from the logs repo have pretty notable differences between the modules.

Reply:
I've added `imbalanced-io.darshan` (from the logs repo) and `sample-dxt-simple.darshan` to the list of logs being tested.